netbox/ipam/api/serializers.py

```diff
@@ -234,10 +234,10 @@ def to_representation(self, instance):
 # IP addresses
 #
 
-class IPAddressInterfaceSerializer(serializers.ModelSerializer):
+class IPAddressInterfaceSerializer(WritableNestedSerializer):
     url = serializers.SerializerMethodField()  # We're imitating a HyperlinkedIdentityField here
-    device = NestedDeviceSerializer()
-    virtual_machine = NestedVirtualMachineSerializer()
+    device = NestedDeviceSerializer(read_only=True)
```

Comment:
This serializer is used for write (POST/PATCH) operations. These fields can't be read-only.